### PR TITLE
docs: Update configuration reference

### PR DESCRIPTION
**Pull request by domdomegg**
- Add undocumented srcPrefix and dstPrefix options
- Better specify types and defaults
- Reorder options to match implementation
- Add example for rawTemplates
- Remove invalid 'inputFormat' option from examples

### CODE REVIEW

**Comment by domdomegg** [comment]
Requesting review from @rtfpessoa

File: README.md [hunk 1]
```diff
@@ -315,25 +314,29 @@ The HTML output accepts a Javascript object with configuration. Possible options
...
   `2500`
 - `maxLineSizeInBlockForComparison`: maximum number os characters of the bigger line in a block to apply comparison,
   default is `200`
-- `maxLineLengthHighlight`: only perform diff changes highlight if lines are smaller than this, default is `10000`
-- `renderNothingWhenEmpty`: render nothing if the diff shows no change in its comparison: `true` or `false`, default is
-  `false`
-- `compiledTemplates`: object with previously compiled templates to replace parts of the html
-- `rawTemplates`: object with raw not compiled templates to replace parts of the html
+- `compiledTemplates`: object (Hogan template values) with previously compiled templates to replace parts of the html,
+  default is `{}`
```
Comment (rtfpessoa) [inline README.md:337]:
```suggestion
- `compiledTemplates`: object ([Hogan.js](https://github.com/twitter/hogan.js/) template values) with previously compiled templates to replace parts of the html, default is `{}`.
  For example: `{ "tag-file-changed": Hogan.compile("<span class="d2h-tag d2h-changed d2h-changed-tag">MODIFIED</span>") }`
```

File: README.md [hunk 2]
```diff
@@ -315,25 +314,29 @@ The HTML output accepts a Javascript object with configuration. Possible options
 
 - `outputFormat`: the format of the output data: `'line-by-line'` or `'side-by-side'`, default is `'line-by-line'`
 - `drawFileList`: show a file list before the diff: `true` or `false`, default is `true`
-- `diffStyle`: show differences level in each line: `word` or `char`, default is `word`
+- `srcPrefix`: add a prefix to all source (before changes) filepaths, default is `''`
+- `dstPrefix`: add a prefix to all destination (after changes) filepaths, default is `''`
```
Comment (rtfpessoa) [inline README.md:318]:
```suggestion
- `srcPrefix`: add a prefix to all source (before changes) filepaths, default is `''`. Should match the prefix used when [generating the diff](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---src-prefixltprefixgt).
- `dstPrefix`: add a prefix to all destination (after changes) filepaths, default is `''`. Should match the prefix used when [generating the diff](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---dst-prefixltprefixgt)
```

**Review by rtfpessoa — CHANGES_REQUESTED** [review]

**Comment by rtfpessoa** [comment]
Hi @domdomegg,

Thanks a lot for your contribution :rocket:

**Pull request merged**